src/components/NewCategory/NewCategory.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from "react";
+import {
+    Dialog,
+    DialogActions,
+    Button,
+    Typography,
+    TextField,
+    Container,
+    makeStyles,
+    Theme,
+    createStyles,
+    Grid,
+} from "@material-ui/core";
+import AddCircleIcon from "@material-ui/icons/AddCircle";
+import CategoryService from "../../services/category.service";
+
+const useStyles = makeStyles((theme: Theme) =>
+    createStyles({
+        padding: {
+            margin: theme.spacing(5),
+        },
+        header: {
+            textTransform: "uppercase",
+            textAlign: "center",
+            marginBottom: theme.spacing(2),
+        },
+        txtField: {
+            width: "100%",
+            marginBottom: theme.spacing(2),
+        },
+        input: {
+            minHeight: theme.spacing(20),
+        },
+    })
+);
+
+const NewCategory: React.FC = () => {
+    const classes = useStyles();
+    const [open, setOpen] = React.useState(false);
+
+    const trigger = () => {
+        setOpen(!open);
+    };
+
+    const [info, setInfo] = React.useState("");
+    const handleChange= (event:React.ChangeEvent<HTMLInputElement>) => {
+        setInfo(event.target.value);
+    };
+    const getInfo = () => {
+        // send data to backend
+        console.log(info);
+        CategoryService.createCategory({
+            name: info,
+            creationTimestamp: Date.now()
+        });
+        setOpen(!open);
+    };
+
+    return (
+        <Container>
+            <AddCircleIcon onClick={trigger}></AddCircleIcon>
+            <Dialog
+                open={open}
+                onClose={trigger}
+                aria-labelledby="form-dialog-title"
+            >
+                <DialogActions>
```

Comment:
Dialog actions is only for buttons (and you dont have to use grid in it)
For textfields use <DialogContent>
For title use <DialogTitle> (no typography required)